File: src/elements/assignment/now-date-to-env.ts
import { BaseSetterAttributes, SetterElement } from "./setter";

export class NowDateToEnv extends SetterElement {
    public static readonly TAG = "now-date-to-env";
    protected attributes = this.attributes as BaseSetterAttributes;

    public getField(): string {
        return this.attributes.field;
    }

    private getAssigned(): string {
        this.converter.addImport("LocalDate");
        this.converter.addImport("java.sql.Date");
        return `Date.valueOf(LocalDate.now())`;
    }

    public getType(): string {
        return "Date";
    }

    public convert(): string[] {
        return [...this.wrapConvert(this.getAssigned())];
    }
}

File: src/main/java/co/uk/stefanpuia/minilang2java/tag/envop/now/NowDateToEnv.java
package co.uk.stefanpuia.minilang2java.tag.envop.now;

import co.uk.stefanpuia.minilang2java.core.TagInit;
import co.uk.stefanpuia.minilang2java.core.model.MinilangTag;
import co.uk.stefanpuia.minilang2java.core.model.VariableType;
import co.uk.stefanpuia.minilang2java.tag.Tag;

@MinilangTag("now-date-to-env")
public class NowDateToEnv extends AbstractNow {

  private final Attributes attributes;

  public NowDateToEnv(final TagInit tagInit) {
    super(tagInit);
    attributes = new NowDateToEnvAttributes(this);
  }

  @Override
  protected Attributes getAttributes() {
    return attributes;
  }

  protected static class NowDateToEnvAttributes extends Attributes {

    protected NowDateToEnvAttributes(final Tag self) {
      super(self);
    }

    @Override
    public VariableType getType() {
      return VariableType.from("java.sql.Date");
    }
  }
}
